presto-parser/src/main/java/com/facebook/presto/sql/tree/ExternalBodyReference.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class ExternalBodyReference
+        extends Node
+{
+    private final Optional<Identifier> identifier;
+    private final Optional<StringLiteral> stringLiteral;
```

Comment:
We are allowing both identifier and string literal in the syntax, but we don't need to store them as is. Instead, I think an `Optional<String> name` would be better, since there is no semantic difference here (i.e., they're not used in expression). See similar example in `CallArgument.java`

Reply:
I kept them separate because otherwise we can't print out the original format in `SqlFormatter`.

Reply:
That's true, but do we need to keep the original format? For example
- `SqlFormatter` renders `array_sort(a)` as `"array_sort"(a)`, 
- Missing routine characteristics in `Create Table` are printed with default values.
So it does not look like we're promising syntactical equivalence with parsing + formatting.

Reply:
Sure, but what's wrong with keeping both? What's the downside?

Reply:
Also, you just allowed `Language` to be an arbitrary identifier. If we follow the same logic of keeping both, don't we want to do that for language as well? So that we can differentiate `SQL` against `"SQL"`?

Reply:
We probably should keep both for language if we want to be strict. I can do that if you insist. The difference is the keyword SQL is basically a special identifier, which string literal is not.

I don't buy your argument. 1) String literals and identifiers have different semantic meaning. For this specific case it doesn't matter. But they are different semantic constructs. You can not change places where asking for one with the other. 2) The constructor already guarantees only one will be present. 3) Sure we can rename them.

If this really bothers you, I'm fine with only supporting string, or only supporting identifier.

Reply:
Yeah - this is a weird one. I can't think of any other place where this situation happens. I think there should be only one "child" of type Node  for the tree node. And you can have an accessor that can give the value of the string literal back.

Reply:
Changed this to only allow `<identifier>` for now. We will see whether there's a need to extend to `StringLiteral` in the future.